src/chains/RouteMatchingProcessorChain.ts

```diff
@@ -0,0 +1,57 @@
+import _ from 'underscore';
+import ProcessorChain, { IRequestMatchingProcessorChain } from './ProcessorChain';
+import { ErrorHandlingRequestProcessor, PathParams } from '../interfaces';
+import { Request } from '..';
+import pathToRegexp from 'path-to-regexp';
+import { StringMap } from '../utils/common-types';
+
+export class RouteMatchingProcessorChain extends ProcessorChain implements IRequestMatchingProcessorChain {
+
+   private readonly _method: string | undefined;
+   private readonly _matcher: RegExp;
+   private readonly _paramKeys: pathToRegexp.Key[] = [];
+
+   public constructor(subprocessors: ErrorHandlingRequestProcessor[], path: PathParams, caseSensitive: boolean = false, method?: string) {
+      super(subprocessors);
+      this._method = method;
+      this._matcher = pathToRegexp(path, this._paramKeys, { sensitive: caseSensitive });
+   }
+
+   public matches(req: Request): boolean {
+      if (this._method !== undefined && req.method !== this._method) {
+         return false;
+      }
+
+      return this._matcher.test(req.path);
+   }
+
+   /**
+    * Only public for the sake of unit testing.
+    */
+   public makeParams(path: string): StringMap {
+      const params: StringMap = {},
+            matches = this._matcher.exec(path);
+
+      if (matches && matches.length > 0) {
+         // the first match returned is the entire URL
+         _.each(matches.slice(1), (v, i) => {
+            const key = this._paramKeys[i];
+
+            if (key && !_.isEmpty(v)) {
+               params[key.name] = v;
+            }
+         });
+      }
+
+      return params;
+   }
+
+   /**
+    * Only public for the sake of unit testing.
+    */
+   public makeSubRequest(req: Request): Request {
```

Comment:
You've probably already considered this, but you could still make this `protected` (and rename to `_makeSubRequest`), and then in the test extend it to expose the `makeSubRequest` method as public:

```typescript
   class TestRouteMatchingProcessorChain extends RouteMatchingProcessorChain {
      public makeSubRequest(req: Request) {
         return this._makeSubRequest(req);
      }
   }
```

Similar to what you did here, for `Router`: https://github.com/silvermine/lambda-express/blob/63cf2601fc2066c000ad46f43bcde8f813f39087/tests/chains/SubRouterProcessorChain.test.ts#L10